### ISSUE TITLE
Should we have a gloo-worker no-op mode for desktop apps?

### ISSUE THREAD

**Issue by gilescope** [rustwasm/gloo#248]
As we move towards 'wasm-first development', I'm wondering if gloo-worker could have a NoOp mode that would work on desktop. I.e. skip the serialisation and just pass the messages in-process. The rational is that once you've got things working for the web, you don't really want to have to re-architect your app to work on desktop. Ideally one could just flip a feature and gloo-worker go into no-op mode and run it all in process. Presumably in this mode it would be able to ditch most dependencies...?

We could create a not-gloo-worker crate and rename it so that the types matched, but that seems less elegant.

**Comment by hamza1311** [comment]
I don't think that is the right solution. gloo-worker is meant to be used to interact with Web Workers.

For your case, you can have a wrapper crate in your project that chooses between the two as needed.
